esphome/components/x9c/x9c.cpp

```diff
@@ -47,18 +47,21 @@ void X9cOutput::setup() {
 
   if (this->initial_value_ <= 0.50) {
     this->trim_value(-101);  // Set min value (beyond 0)
-    this->trim_value(static_cast<uint32_t>(roundf(this->initial_value_ * 100)));
+    this->trim_value(static_cast<int32_t>(roundf(this->initial_value_ * 100)));
   } else {
     this->trim_value(101);  // Set max value (beyond 100)
-    this->trim_value(static_cast<uint32_t>(roundf(this->initial_value_ * 100) - 100));
+    this->trim_value(static_cast<int32_t>(roundf(this->initial_value_ * 100) - 100));
   }
   this->pot_value_ = this->initial_value_;
   this->write_state(this->initial_value_);
```

Comment:
```suggestion
  this->pot_value_ = static_cast<int32_t>(roundf(this->initial_value_ * 100));
```
We don't have to call write state as the value was already trimmed, I also makes more sense to store the value * 100